### PR TITLE
[FIX] website: prevent font from reverting after custom font upload

### PR DESCRIPTION
**Pull request by paru-odoo**
Steps to reproduce:

1. Go to Website -> Edit -> Theme section.
2. Choose a predefined font family for the paragraph, such as "Arvo".
4. Return to the font family dropdown and click on "Add a Custom Font".
Notice that the selected font reverts to the "default/system" font.

Issue:
The issue occurs because when a predefined "we-button" is selected from the dropdown menu(we-button with value i.e. data attributes defined), its value is passed to widgetValue. However, when the "Add a Custom Font" button is clicked(which does not have value i.e. data attributes are not defined), the "we-button" value changes to an empty string, which is then passed into widgetValue. As a result, it triggers refreshBundle with the empty string and resets to the default value.

Solution:
This PR ensures that `_refreshBundles` is not triggered until the widgetValue has a valid (non-empty) value. This prevents the font from being reverted to the default.

task-4373983

### CODE REVIEW

File: addons/website/static/src/js/editor/snippets.options.js [hunk 1]
```diff
@@ -1028,7 +1028,7 @@ options.Class.include({
                 }
         }
 
-        if (params.reload || params.noBundleReload) {
+        if (params.reload || params.noBundleReload || widgetValue?.length === 0) {
```
Comment (shsa-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:1031]: I dont think we need to check all those conditions above as well, right do we need to process the switch case if widgetValue is not there ?
We can simply return from the method from the first line.
what is your view on this

File: addons/website/static/tests/tours/font_familty.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/** @odoo-module **/
+import wTourUtils from '@website/js/tours/tour_utils';
+
+wTourUtils.registerWebsitePreviewTour('website_font_family', {
+    test: true,
+    url: '/',
+    edition: true,
```
Comment (shsa-odoo) [inline addons/website/static/tests/tours/font_familty.js:7]: double quotes everywhere.

File: addons/website/static/tests/tours/font_familty.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        run: "click",
+    },
+    {
+        content: "Wait for the modal to open and then refresh",
+        trigger: "button.btn-secondary",
+        run: function () {
+            setTimeout(() => {
+                this.anchor.click();
+            }, 2000);
+        },
```
Comment (shsa-odoo) [inline addons/website/static/tests/tours/font_familty.js:75]: Please explain what was actual need to do this, was adding the selector to trigger not working ?

**Review by shsa-odoo — COMMENTED** [review]
Hello @paru-odoo Few comments needs addressing. Thanks for your work.

File: addons/website/static/tests/tours/font_familty.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                setTimeout(() => {
+                    this.anchor.click();
+                }, 2000);
+            },
+        },
+        {
+            content:
+                "Check that 'Lato-BlaIta-webfont' font family is still applied and not reverted",
+            trigger:
+                "we-select[data-variable='headings-font'] we-toggler[style*='font-family: Lato-BlaIta-webfont;']",
```
Comment (msh-odoo) [inline addons/website/static/tests/tours/font_familty.js:91]: Tour is too specific for this case, maybe we should write tour with generic case instead of following the same steps given in task, we should test reload/_refreshBundles should not be called when we select Add Custom Font ```we-button```

Comment (shsa-odoo) [reply addons/website/static/tests/tours/font_familty.js:91]: Hello @msh-odoo "The current [steps](https://github.com/odoo/odoo/pull/190329/files#diff-7fc5f8398accb6d9385a64f6e66ef60342d824794f0e0cdbb474c9aa403eff27R68-R91) are designed specifically to test that reload/_refreshBundles are not called, as the 'Lato-BlaIta-webfont' font family remains applied and does not revert. This ensures these methods were not triggered, since triggering them would have caused the final validation step to fail."

File: addons/website/static/src/js/editor/snippets.options.js (hunk 1 above)
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:1031]: 1. Why not writing that `widgetValue === ""` instead of this `widgetValue?.length === 0`? :roll_eyes: 
2. Why would having an empty string value be related to not refreshing the bundle. It feels like you are fixing generically a very specific case.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
> When the "Add Custom Font" button is clicked, the widgetValue is set to an empty string.

And is that normal? Should not the bug fix be about avoiding that?

File: addons/website/static/src/js/editor/snippets.options.js (hunk 1 above)
Comment (msh-odoo) [reply addons/website/static/src/js/editor/snippets.options.js:1031]: @qsm-odoo We can do changes for ```widgetValue``` comparison but this is not specific case, this can be produced when you click on we-button without data attributes, for example select any font-family from dropdown and then click on Add a Custom Font, it will remove the previous value set on dropdown, so the problem is Add a Custom Font does not have value, we have written tour steps in that way only.

Comment (qsm-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: I don't understand your answer @msh-odoo -> how is this not a specific case which is to be fixed here?

Comment (paru-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: @qsm-odoo This video explains the following behavior: When a predefined "we-button" is selected from the dropdown menu(we-button with value i.e. data attributes defined), its value is passed to widgetValue. However, when the "Add a Custom Font" button is clicked(which does not have value i.e. data attributes are not defined), the "we-button" value changes to an empty string, which is then passed into widgetValue. As a result, it triggers refreshBundle with the empty string and resets to the default value.
https://www.loom.com/share/4cbea11a28e246ad9f3d29f55bae4d5d?sid=a50bf8c9-1b2b-41ad-9714-d145c219ec01

Comment (qsm-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: > when the "Add a Custom Font" button is clicked

That looks like a **specific case** to me -> why are we saying, **generically for all buttons using this option method, in all potential custo**, "if there is no value, then the bundle should not be refreshed"?

Comment (paru-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: @qsm-odoo, I have updated the commit message accordingly.

Comment (qsm-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: @paru-odoo I think you forgot to push your adaptation of the code accordingly

Comment (paru-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: @qsm-odoo , Are you referring to ```WidgetValue === ""```? If so, I have already updated the code accordingly. Or are you referring to something else?

Comment (msh-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: @qsm-odoo I got your point, We will check a better solution instead of checking empty string ```widgetValue === ""``` or we will give right justification, Thanks for your review.

File: addons/website/static/tests/tours/font_family.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        trigger: "we-select[data-variable='headings-font'] .o_we_add_font_btn",
+        run: "click",
+    },
+    {
+        content: "Wait for the modal to open and then refresh",
+        trigger: "button.btn-secondary",
+        run: function () {
+            setTimeout(() => {
+                this.anchor.click();
+            }, 2000);
```
Comment (msh-odoo) [inline addons/website/static/tests/tours/font_family.js:40]: Can we avoid this setTimeout?

**Review by qsm-odoo — APPROVED** [review]
Seems good, not really stable but... :shrug: I hope the test is nice :+1: 

@robodoo r+

**Comment by msh-odoo** [comment]
> Seems good, not really stable but... 🤷 I hope the test is nice 👍
> 
> @robodoo r+

Missed it due to many conversation on this PR ;)
